swift/src/Ice/LocalException.swift

```diff
@@ -1,1030 +1,47 @@
-    override open func ice_print() -> String {
-        return _FeatureNotSupportedExceptionDescription
-    }
-}
-
-/// This exception indicates a failure in a security subsystem, such as the IceSSL plug-in.
-open class SecurityException: LocalException {
-    /// The reason for the failure.
-    public var reason: String = ""
-
-    public required init() {
-        super.init()
-    }
-
-    public init(reason: String, file: String = #file, line: Int32 = #line) {
-        self.reason = reason
-        super.init(file: file, line: line)
-    }
-
-    /// Returns the Slice type ID of this exception.
-    ///
-    /// - returns: `String` - the Slice type ID of this exception.
-    override open class func ice_staticId() -> String {
-        return "::Ice::SecurityException"
-    }
-
-    /// Returns a string representation of this exception
-    ///
-    /// - returns: `String` - The string representaton of this exception.
-    override open func ice_print() -> String {
-        return _SecurityExceptionDescription
-    }
-}
-
-/// This exception indicates that an attempt has been made to change the connection properties of a fixed proxy.
-open class FixedProxyException: LocalException {
-    /// Returns the Slice type ID of this exception.
-    ///
-    /// - returns: `String` - the Slice type ID of this exception.
-    override open class func ice_staticId() -> String {
-        return "::Ice::FixedProxyException"
-    }
-
-    /// Returns a string representation of this exception
-    ///
-    /// - returns: `String` - The string representaton of this exception.
-    override open func ice_print() -> String {
-        return _FixedProxyExceptionDescription
+/// Base class for Ice local exceptions.
+open class LocalException: Exception, CustomStringConvertible {
+    public let message: String
+    public let file: String
+    public let line: Int32
+    private let cxxDescription: String?
+
+    /// A textual representation of this Ice exception.
+    public var description: String {
+        cxxDescription ?? "\(file):\(line) \(ice_id()) \(message)"
+    }
+
+    /// Creates a LocalException.
+    /// - Parameters:
+    ///   - message: The exception message.
+    ///   - file: The file where the exception was thrown.
+    ///   - line: The line where the exception was thrown.
+    public init(_ message: String, file: String = #file, line: Int32 = #line) {
+        self.message = message
+        self.file = file
+        self.line = line
+        self.cxxDescription = nil
+    }
+
+    /// Creates a LocalException from an Ice C++ local exception.
+    /// - Parameters:
+    ///   - message: The exception message.
+    ///   - cxxDescription: The C++ exception description.
+    ///   - file: The file where the exception was thrown.
+    ///   - line: The line where the exception was thrown.
+    public required init(message: String, cxxDescription: String, file: String, line: Int32) {
```

Comment:
"required" is required for creating instances by name (see NSClassFromString code).